blockchain/chain/src/store/chain_store.rs

```diff
@@ -59,21 +71,16 @@ where
         };
         Ok(self.tip_index.put(&meta)?)
     }
-    /// weight
-    pub fn weight(&self, _ts: &Tipset) -> Result<BigUint, Error> {
-        // TODO
-        Ok(BigUint::from(0 as u32))
-    }
 
     /// Writes genesis to blockstore
-    pub fn set_genesis(&self, header: BlockHeader) -> Result<(), Error> {
+    pub fn set_genesis(&mut self, header: BlockHeader) -> Result<(), Error> {
```

Comment:
why do you need to change to mutable?

Reply:
Because `persist_headers` needs a mutable reference.